tests/queries/0_stateless/03277_async_insert.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+set -e
+
+
+rm -rf "${CLICKHOUSE_USER_FILES_UNIQUE}"
+mkdir -p "${CLICKHOUSE_USER_FILES_UNIQUE}"
+
+
+FILE_CSV="${CLICKHOUSE_USER_FILES_UNIQUE}/03277.csv"
+FILE_ARROW="${CLICKHOUSE_USER_FILES_UNIQUE}/03277.arrow"
+
+
+$CLICKHOUSE_CLIENT -q "
+    SET async_insert = 1;
+    SELECT '${FILE_CSV}';
+    CREATE TABLE t0 (c0 Int) ENGINE = File(CSV);
+    INSERT INTO TABLE t0 (c0) VALUES (1);
+    INSERT INTO TABLE FUNCTION file('$FILE_CSV', 'CSV', 'c0 Int') SELECT c0 FROM t0;
+    INSERT INTO TABLE t0 (c0) FROM INFILE '$FILE_CSV' FORMAT CSV;
+"
+$CLICKHOUSE_CLIENT -q "SELECT * from t0"
+$CLICKHOUSE_CLIENT -q "DROP TABLE t0"
+
+$CLICKHOUSE_CLIENT -q "
+    SET async_insert = 1;
+    SELECT '${FILE_ARROW}';
+    CREATE TABLE t0 (c0 Int) ENGINE = Join(ANY, INNER, c0);
```

Comment:
Is issue specific to `Join` table engine? If yes, what particularly cases it?

Reply:
No. It turns out that it related to the empty set of insertion data.

Reply:
We created pipeline and destroyed it with no execution at all.